lib/std/zig/parser_test.zig

```diff
@@ -2846,10 +2876,25 @@ fn testTransform(source: []const u8, expected_source: []const u8) !void {
         const result_source = try testParse(source, &failing_allocator.allocator, &anything_changed);
         if (!mem.eql(u8, result_source, expected_source)) {
             warn("\n====== expected this output: =========\n", .{});
-            warn("{}", .{expected_source});
+            printWithVisibleNewlines(expected_source);
             warn("\n======== instead found this: =========\n", .{});
-            warn("{}", .{result_source});
+            printWithVisibleNewlines(result_source);
             warn("\n======================================\n", .{});
+
+            const diff_index = mem.diffIndex(u8, result_source, expected_source);
+
+            var diff_line_number: usize = 1;
+            for (expected_source[0..diff_index]) |value| {
+                if (value == '\n') diff_line_number += 1;
+            }
+            warn("First difference occurs on line {}:\n", .{diff_line_number});
+
+            warn("expected:\n", .{});
+            printIndicatorLine(expected_source, diff_index);
+
+            warn("found:\n", .{});
+            printIndicatorLine(result_source, diff_index);
+
```

Comment:
I think this would be really useful in `std.testing`.

Reply:
Should I make that a part of this pull request?

Reply:
Might as well. I think it would also cover the use case of #4882.